test/lightning/runs/prom_ex_plugin/impeded_project_helper_test.exs

```diff
@@ -0,0 +1,485 @@
+          workflow_id: "workflow-id-2",
+          inprogress_runs_count: 5,
+          project_concurrency: 10,
+          workflow_concurrency: 6
+        }
+      ]
+
+      refute ImpededProjectHelper.project_has_unused_concurrency?(stats)
+    end
+
+    test "false if sum of runs > proj concurrency - ignores workflows" do
+      stats = [
+        %{
+          project_id: "project-id",
+          workflow_id: "workflow-id-1",
+          inprogress_runs_count: 6,
+          project_concurrency: 10,
+          workflow_concurrency: 6
+        },
+        %{
+          project_id: "project-id",
+          workflow_id: "workflow-id-2",
+          inprogress_runs_count: 5,
+          project_concurrency: 10,
+          workflow_concurrency: 6
+        }
+      ]
+
+      refute ImpededProjectHelper.project_has_unused_concurrency?(stats)
+    end
+
+    test "true if runs < proj concurrency & workflow with capacity == 1" do
+      stats = [
+        %{
+          project_id: "project-id",
+          workflow_id: "workflow-id-1",
+          inprogress_runs_count: 6,
+          project_concurrency: 10,
+          workflow_concurrency: 6
+        },
+        %{
+          project_id: "project-id",
+          workflow_id: "workflow-id-2",
+          inprogress_runs_count: 3,
+          project_concurrency: 10,
+          workflow_concurrency: 6
+        }
+      ]
+
+      assert ImpededProjectHelper.project_has_unused_concurrency?(stats)
+    end
+
+    test "true if runs < proj concurrency & workflows with capacity > 1" do
+      stats = [
+        %{
+          project_id: "project-id",
+          workflow_id: "workflow-id-1",
+          inprogress_runs_count: 5,
+          project_concurrency: 20,
+          workflow_concurrency: 6
+        },
+        %{
+          project_id: "project-id",
+          workflow_id: "workflow-id-2",
+          inprogress_runs_count: 6,
+          project_concurrency: 20,
+          workflow_concurrency: 6
+        },
+        %{
+          project_id: "project-id",
+          workflow_id: "workflow-id-1",
+          inprogress_runs_count: 1,
+          project_concurrency: 20,
+          workflow_concurrency: 6
+        }
+      ]
+
+      assert ImpededProjectHelper.project_has_unused_concurrency?(stats)
+    end
+
+    test "false if sum of runs < p concurrency workflows with capacity == 0" do
+      stats = [
+        %{
+          project_id: "project-id",
+          workflow_id: "workflow-id-1",
+          inprogress_runs_count: 4,
+          project_concurrency: 10,
+          workflow_concurrency: 4
+        },
+        %{
+          project_id: "project-id",
+          workflow_id: "workflow-id-2",
+          inprogress_runs_count: 4,
+          project_concurrency: 10,
+          workflow_concurrency: 4
+        }
+      ]
+
+      refute ImpededProjectHelper.project_has_unused_concurrency?(stats)
+    end
+  end
+
+  describe "project_has_unused_concurrency?/1 - project concurrency is nil" do
```

Comment:
@taylordowns2000 These tests also cover the effect of the project-level concurrency.